### PR TITLE
ci: use checkout with fetch-depth 0

### PR DESCRIPTION
**Pull request by afuetterer**
See: https://github.com/PyPSA/PyPSA/pull/954

### CODE REVIEW

File: .github/workflows/gh-release.yml [hunk 1]
```diff
@@ -10,9 +10,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-
-      - name: Fetch all history for all tags and branches (necessary for setuptools_scm)
-        run: git fetch --prune --unshallow
+        with:
+          fetch-depth: 0 # Fetch all history for all tags and branches (necessary for setuptools_scm)
```
Comment (afuetterer) [inline .github/workflows/gh-release.yml:14]: Actually I think you don't need all the commits here, right?
Maybe it was copy-and-pasted here?

**Comment by afuetterer** [comment]
Two questions.

Regarding gh-release workflow:
- In my opinion this line is redundant, because the workflow only runs, when a tag is pushed:
https://github.com/PyPSA/linopy/blob/20a6dcdcd8dc4221c8f417e219075da0524385f7/.github/workflows/gh-release.yml#L19
- Can we remove it?

Regarding deploy workflow:
- The workflow runs on push, so every commit builds the package, but only on a tag it is deployed.
- Is this intended? Or should deploy run on tag only?
https://github.com/PyPSA/linopy/blob/20a6dcdcd8dc4221c8f417e219075da0524385f7/.github/workflows/deploy.yml#L37
- maybe you could use [build-and-inspect-python-package](https://github.com/hynek/build-and-inspect-python-package) in the ci workflow? It tests if the package is "buildable" and runs a few checks on the files, metadata etc.

**Comment by FabianHofmann** [comment]
thanks for the PR and raising the issue @afuetterer, I think @lkstrp knows :)

**Comment by afuetterer** [comment]
Example of "baipp" workflow: https://github.com/hynek/structlog/blob/main/.github/workflows/ci.yml

Example output: https://github.com/hynek/build-and-inspect-python-package/actions/runs/9817023209

**Comment by afuetterer** [comment]
What does that mean for this PR?

**Comment by lkstrp** [comment]
You can apply everything proposed in [here](https://github.com/PyPSA/linopy/pull/313#issuecomment-2213139573).

File: .github/workflows/gh-release.yml (hunk 1 above)
Comment (afuetterer) [reply .github/workflows/gh-release.yml:14]: How about this one? It should work without pulling all commits, right?

Comment (FabianHofmann) [reply .github/workflows/gh-release.yml:14]: @lkstrp let's try it out?

Comment (lkstrp) [reply .github/workflows/gh-release.yml:14]: Think so, let's go. 
But I would also merge `gh-release.yml` with `deploy.yml`.

**Comment by afuetterer** [comment]
> @afuetterer ready to merge?

Yes. Will open another PR soon.

File: .github/workflows/gh-release.yml (hunk 1 above)
Comment (afuetterer) [reply .github/workflows/gh-release.yml:14]: Want to do it in this PR or another one?

**Comment by FabianHofmann** [comment]
great thanks @afuetterer

**Pull request merged**